pkg/postgres/pgadmin/admin_utils.go

```diff
@@ -107,20 +107,63 @@ func CheckIfDBExists(config *pgxpool.Config, dbName string) bool {
 	return exists
 }
 
+// AnalyzeDatabase - runs ANALYZE on the database named dbName
+func AnalyzeDatabase(config *pgxpool.Config, dbName string) error {
+	log.Debugf("Analyze - %q", dbName)
+
+	// Connect to different database for admin functions
+	connectPool := GetReplicaPool(config, dbName)
+	// Close the admin connection pool
+	defer connectPool.Close()
+
+	// Create a context with a timeout
+	ctx, cancel := context.WithTimeout(context.Background(), postgresQueryTimeout)
+	defer cancel()
+
+	_, err := connectPool.Exec(ctx, "ANALYZE")
+
+	return err
+}
+
 // GetAdminPool - returns a pool to connect to the admin database.
 // This is useful for renaming databases such as a restore to active.
-func GetAdminPool(config *pgxpool.Config) *pgxpool.Pool {
-	var err error
-	var postgresDB *pgxpool.Pool
-
+// THIS POOL SHOULD BE CLOSED ONCE ITS PURPOSE HAS BEEN FULFILLED.
+func GetAdminPool(pgConfig *pgxpool.Config) *pgxpool.Pool {
 	// Clone config to connect to template DB
-	tempConfig := config.Copy()
+	tempConfig := pgConfig.Copy()
 
 	// Need to connect on a static DB so we can rename the used DBs.
 	tempConfig.ConnConfig.Database = AdminDB
 
+	postgresDB := getPool(tempConfig)
+
+	log.Debugf("Got connection pool for database %q", AdminDB)
+	return postgresDB
+}
+
+// GetReplicaPool - returns a connection pool for the specified database replica.
+// THIS POOL SHOULD BE CLOSED ONCE ITS PURPOSE HAS BEEN FULFILLED.
+func GetReplicaPool(pgConfig *pgxpool.Config, replica string) *pgxpool.Pool {
```

Comment:
[nit] Are there better options to name this database except "replica"? It will be inevitably confusing in the future when the actual replicas (streaming/logical replica instances) will be involved.

Reply:
@erthalion yeah that has been bugging me a little as I've been working on the rollback/upgrade stuff.  I think I will solve this there.  Either as part of ROX-11677 or some other subtask under ROX-9884.  I'm leaning toward changing replica to clone as that probably more closely resembles what these things are.